napari_builtins/_qt/features_table.py

```diff
@@ -0,0 +1,544 @@
+
+    def createEditor(self, parent, option, index):
+        proxy_model = index.model()
+        source_model = proxy_model.sourceModel()
+        source_index = proxy_model.mapToSource(index)
+        col = source_index.column()
+
+        dtype = source_model.df.dtypes.iat[col - 1]
+
+        if pd.api.types.is_categorical_dtype(dtype):
+            editor = QComboBox(parent)
+            categories = source_model.df.iloc[:, col - 1].cat.categories
+            editor.addItems([str(c) for c in categories])
+            # allow arrow keys selection
+            editor.setFocusPolicy(Qt.StrongFocus)
+
+            # force editor to open on first click, otherwise we need 2 clicks
+            QTimer.singleShot(0, editor.showPopup)
+            return editor
+
+        return super().createEditor(parent, option, index)
+
+    def setEditorData(self, editor, index):
+        if isinstance(editor, QComboBox):
+            value = index.model().data(index, Qt.EditRole)
+            i = editor.findText(value)
+            if i >= 0:
+                editor.setCurrentIndex(i)
+        else:
+            super().setEditorData(editor, index)
+
+    def setModelData(self, editor, model, index):
+        if isinstance(editor, QComboBox):
+            source_index = model.mapToSource(index)
+            source_model = model.sourceModel()
+            source_model.setData(
+                source_index, editor.currentText(), Qt.EditRole
+            )
+        else:
+            super().setModelData(editor, model, index)
+
+
+class BoolFriendlyProxyModel(QSortFilterProxyModel):
+    """Sort proxy model that handles booleans correctly."""
+
+    def lessThan(self, left, right):
+        left_data = self.sourceModel().data(left, Qt.EditRole)
+        right_data = self.sourceModel().data(right, Qt.EditRole)
+
+        # ensure booleans compare as expected
+        if isinstance(left_data, bool) and isinstance(right_data, bool):
+            return not left_data and right_data
+
+        return super().lessThan(left, right)
+
+
+class PandasView(QTableView):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        # view/model setup with a proxy for sorting and filtering
+        proxy_model = BoolFriendlyProxyModel()
+        proxy_model.setSourceModel(PandasModel())
+        proxy_model.setSortCaseSensitivity(Qt.CaseInsensitive)
+        self.setModel(proxy_model)
+        self.setSortingEnabled(True)
+        # do not auto sort using index on startup
+        self.horizontalHeader().setSortIndicator(-1, Qt.AscendingOrder)
+        # disable vertical header (since we duplicate it as a column)
+        self.verticalHeader().setVisible(False)
+        # delegate which provides comboboxes for editing categorical values
+        self.setItemDelegate(DelegateCategorical())
+        # enable selection and editing
+        self.setSelectionBehavior(QTableView.SelectItems)
```

Comment:
```suggestion
        self.setSelectionBehavior(QTableView.SelectionBehavior.SelectItems)
```

And more in this class